src/shared/components/wallet/Wallet.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+
+import createWallet from 'idm-wallet';
+import { CSSTransition } from 'react-transition-group';
+
+import WalletContent from './wallet-content/WalletContent';
+import SetupLocker from '../setup-locker/SetupLocker';
+import LockScreen from '../lock-screen';
+
+import styles from './Wallet.css';
+
+class Wallet extends Component {
+    getLock = undefined;
+
+    state = {
+        wallet: undefined,
+    };
+
+    componentDidMount() {
+        createWallet().then((wallet) => {
+            wallet.locker.onLockedChange(this.handleLockedChanged);
+            this.getLock = wallet.locker.getLock.bind(wallet.locker);
+
+            this.setState({ wallet });
+        });
+    }
+
+    render() {
+        const { wallet } = this.state;
+
+        if (wallet) {
+            return this.renderWallet();
+        }
+
+        return <div>No Wallet</div>;
+    }
+
+    renderWallet = () => {
+        const { wallet } = this.state;
+        const { locker } = wallet;
+
+        if (locker.isPristine()) {
+            return <SetupLocker locker={ locker } onComplete={ this.handleSetupLockerComplete } />;
+        }
+
+        const isLocked = locker.isLocked();
+
+        return (
+            <div className={ styles.parent }>
+                <CSSTransition classNames={ styles.appScreen } in={ isLocked } mountOnEnter unmountOnExit component={ null } timeout={ 1800 }>
+                    <LockScreen getLock={ this.getLock } unmounting={ !isLocked } />
+                </CSSTransition>
+                <CSSTransition classNames="none" in={ !isLocked } mountOnEnter unmountOnExit component={ null } timeout={ 2000 }>
```

Comment:
I wouldn't use `CSSTransition` just to mount/unmount this component. I would use [`onExited` callback](http://reactcommunity.org/react-transition-group/css-transition#CSSTransition-prop-onExited) on the previous `CSSTransition` and then would update some state variable that allows this component to render.

Reply:
I like using `CSSTransition` for the uniformity and easing the mounting, unmounting and use of timeouts. I feel the same way as with specifying classes: unless there is a performance drawback, I prefer having it implemented as it is.

If we end up deciding to not unmount the wallet content to maintain the state of the application during the locked state, the choice of implementation now is indifferent.

@satazor do you want to come unblock these discussions?